File: src/services/working-hours.service.js
/**
 * Working Hours Service - Working hours management API calls
 */

import { apiClient } from '../core/api-client.js';

export class WorkingHoursService {
  async getWorkingHours() {
    return await apiClient.get('/admin/working-hours');
  }

  async updateWorkingHours(workingHoursData) {
    return await apiClient.put('/admin/working-hours', workingHoursData);
  }
}

export const workingHoursService = new WorkingHoursService();

File: backend/app/services/working_hours_service.py
"""Working Hours Service - Business logic for working hours management"""

from sqlalchemy.orm import Session
from .. import crud, schemas


class WorkingHoursService:
    """Service for working hours operations"""
    
    @staticmethod
    def get_working_hours(db: Session) -> schemas.WorkingHoursResponse:
        """Get all working hours"""
        items = crud.list_working_hours(db)
        return schemas.WorkingHoursResponse(items=items)
    
    @staticmethod
    def update_working_hours(
        db: Session,
        payload: schemas.WorkingHoursUpdate
    ) -> schemas.WorkingHoursResponse:
        """Update working hours"""
        items = crud.update_working_hours(db, payload.items)
        return schemas.WorkingHoursResponse(items=items)
